kapitan/cli.py

```diff
@@ -438,13 +411,15 @@ def secret_reveal(args, ref_controller):
     file_name = args.file
     if file_name is None:
         fatal_error('--file is required with --reveal')
-    if file_name == '-':
-        # TODO deal with RefHashMismatchError or KeyError exceptions
-        out = revealer.reveal_raw_file(None)
-        sys.stdout.write(out)
-    elif file_name:
-        for rev_obj in revealer.reveal_path(file_name):
-            sys.stdout.write(rev_obj.content)
+    try:
+        if file_name == '-':
+            out = revealer.reveal_raw_file(None)
+            sys.stdout.write(out)
+        elif file_name:
+            for rev_obj in revealer.reveal_path(file_name):
+                sys.stdout.write(rev_obj.content)
+    except (RefHashMismatchError, KeyError):
```

Comment:
What would throw KeyError here?

Reply:
I added it because it was mentioned in the # TODO comment.
After looking at the code a bit I think it might be thrown by
```
self.ref_controller[tag]
``` in `_reveal_replace_match` reachable in the Revealer (?).
If it's not that, I could remove it.

Reply:
cool, was curious